source/production/F0.Minesweeper.Logic/Minefield.cs

```diff
@@ -54,6 +35,15 @@ internal Minefield(uint width, uint height, uint mineCount, MinefieldFirstUncove
 			minefieldOptions.GenerationOption)
 		{ }
 
+		internal Minefield(uint width, uint height, uint mineCount, IMinelayer mineplacer)
+		{
+			this.width = width;
+			this.height = height;
+			this.mineCount = mineCount;
+			this.mineplacer = mineplacer;
+			minefield = new List<Cell>();
+		}
+
 		public IGameUpdateReport Uncover(uint x, uint y) => Uncover(new Location(x, y));
 		public IGameUpdateReport Uncover(Location location)
```

Comment:
Do we want to trust that the location that's passed here is valid?

Reply:
You are right. I added a check `if(!GetAllLocations().Contains(location))` in the `Uncover` method before calling `UncoverCells(location)` and throwing an `ArgumentException` in the case the Location is invalid. What do you think? 🤔

Reply:
That sounds like a great solution :)